### ISSUE TITLE
CosineProximity loss function throwing error

### ISSUE THREAD

**Issue by kedevked** [tensorflow/tfjs#1385]
To get help from the community, we encourage using Stack Overflow and the [`tensorflow.js`](https://stackoverflow.com/questions/tagged/tensorflow.js) tag.

#### TensorFlow.js version
1.0.1
#### Browser version
chrome 71
#### Describe the problem or feature request
Cosine proximity loss function is throwing a shape mismatch error 
#### Code to reproduce the bug / link to feature request

```
(async() => {
  model = tf.sequential();
model.add(tf.layers.dense({units: 1, inputShape: [1], activation: 'relu', kernelInitializer:'ones'}));

model.compile({loss: 'cosineProximity', optimizer: 'sgd'});

    // Train
    const logs = await model.fit(tf.ones([4, 1]), tf.ones([4, 1]), {epochs: 10});
    console.log(logs)
})()
```

**Comment by kaesve** [comment]
I took a few minutes looking at this, and I think the issue is that `cosineProximity` actually isn't a loss function, but a metric (see https://js.tensorflow.org/api/latest/#metrics.cosineProximity ). Metrics aren't meant to be used as error functions and don't have to be differentiable, I think.

Instead of `cosineProximity` you can use `cosineDistance` like so:

```
model.compile({loss: tf.losses.cosineDistance, optimizer: 'sgd'});,
```

You can't use it by name (eg `loss: "cosineDistance"`) and I'm not sure why. It feels a bit backwards that using  `loss: "cosineProximity"` compiles but breaks during fitting, while the cosineDistance works but can't be found by name.

**Comment by bileschi** [comment]
Looks like this is a holdover from @keras-team/keras here:
https://keras.io/losses/

Agreed TFJS should have string-level access to cosineDistance loss.
It would need to be added here:
https://github.com/tensorflow/tfjs-layers/blob/1f32448ee4f0b15eccdbe45bbf4d259b5a4ef8d8/src/losses.ts#L305

**Comment by bileschi** [comment]
cc @caisq & @davidsoergel for opinion on whether this should be added.
It would also need to be added to the keras_format spec here:
https://github.com/tensorflow/tfjs-layers/blob/master/src/keras_format/loss_config.ts

**Comment by FremyCompany** [comment]
Hey, I just wanted to point out that the function already exists but is incorrectly implemented. I fixed this issue by myself now, and that the fix can be merged via https://github.com/tensorflow/tfjs-layers/pull/499

**Comment by davidsoergel** [comment]
The confusion here is that tfjs-core provides some 'losses' (but no 'metrics' per se), and tfjs-layers provides a partly-overlapping set of 'losses' and 'metrics'.  Only the tfjs-layers ones are expected to be referenceable by string names such as `cosineProximity`.  The allowed names are defined by the Keras API (e.g., https://keras.io/losses/).  Many of the tfjs-core losses are actually metrics (in the Keras terminology) so they are re-exported as tfjs-layers *metrics*.

Thus, while this is all very confusing, I think we should *not* add `cosineDistance` to our representation of the Keras spec, since @keras-team/keras doesn't support it.

In combination with tensorflow/tfjs-layers#499, I think that means this bug is now fixed.
